java-frontend/src/test/files/sym/ConditionalOperator.java

```diff
@@ -0,0 +1,10 @@
+import java.util.List;
+public class App {
+  public class Foo<F extends List<Object>> {
+  }
+  public class Bar<G extends List<Object>> extends Foo<G> {
+  }
+  private <H extends List<Object>> Foo fun() {
+    return true ? new Bar() : new Foo<List<Object>>(); // resolving type of this expression leads to a failing invariant in LeastUpperBound computation
```

Comment:
comment should be updated to reflect how it will behave, or at least not tell that it fails.